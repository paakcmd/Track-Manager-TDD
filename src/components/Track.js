import React, { Component } from 'react';
import { createTrack } from '../actions/track';
import { connect } from 'react-redux';

export class Track extends Component {
  display(track) {
    if(track){
      if (track.length > 0) {
        return track.map(trackLine => (
          <tr key={trackLine.time}>
            <td>{trackLine.time}</td>
            <td>{trackLine.event}</td>
          </tr>
        ));
      } else {
        return <tr id="emptyList"><td>Empty</td><td>Empty</td></tr>
      }
    }
  }
  render() {
    const track = this.props.track;
    return (
      <div>
        <input
          className="inputFile"
          name="file"
          id="file"
          type="file"
          accept=".txt"
          onChange={e => this.props.createTrack(e.target.files[0])}
        />
        <label htmlFor="file">Choose a file</label>
        <table>
          <tbody>
            <tr>
              <th>Time</th>
              <th>Event</th>
            </tr>
            {this.display(track)}
          </tbody>
        </table>
      </div>
    );
  }
}

export default connect(
  state => {
    return { track: state };
  },
  { createTrack }
)(Track);
