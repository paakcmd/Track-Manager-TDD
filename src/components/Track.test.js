import React from 'react';
import { shallow } from 'enzyme';
import { Track } from './Track';

describe('Track', () => {
  const mockCreateTrack = jest.fn();
  const props = { createTrack: mockCreateTrack }
  const track = shallow(<Track {...props} />);
  it('renders properly', () => {
    expect(track).toMatchSnapshot();
  })

  describe('when user upload file', () => {
    beforeEach(() => {
      track.find('#input').simulate('change', {target: {files: [0: 'test']}})
    })
    it('send file to the reducer', () => {
      expect(mockCreateTrack).toHaveBeenCalledWith(0);
    })
  })
})
