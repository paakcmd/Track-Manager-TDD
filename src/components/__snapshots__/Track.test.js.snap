// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Track
    createTrack={[Function]}
    track={
        Array [
            Object {
              "event": "test",
              "time": "9.00",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          accept=".txt"
          className="inputFile"
          id="file"
          name="file"
          onChange={[Function]}
          type="file"
/>,
        <label
          htmlFor="file"
>
          Choose a file
</label>,
        <table>
          <tbody>
                    <tr>
                              <th>
                                        Time
                              </th>
                              <th>
                                        Event
                              </th>
                    </tr>
                    <tr>
                              <td>
                                        9.00
                              </td>
                              <td>
                                        test
                              </td>
                    </tr>
          </tbody>
</table>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "accept": ".txt",
          "className": "inputFile",
          "id": "file",
          "name": "file",
          "onChange": [Function],
          "type": "file",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Choose a file",
          "htmlFor": "file",
        },
        "ref": null,
        "rendered": "Choose a file",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tbody>
            <tr>
                        <th>
                                    Time
                        </th>
                        <th>
                                    Event
                        </th>
            </tr>
            <tr>
                        <td>
                                    9.00
                        </td>
                        <td>
                                    test
                        </td>
            </tr>
</tbody>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <th>
                                Time
                </th>
                <th>
                                Event
                </th>
</tr>,
              Array [
                <tr>
                  <td>
                                    9.00
                  </td>
                  <td>
                                    test
                  </td>
</tr>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    Time
</th>,
                  <th>
                    Event
</th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Time",
                  },
                  "ref": null,
                  "rendered": "Time",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Event",
                  },
                  "ref": null,
                  "rendered": "Event",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": "9.00",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    9.00
</td>,
                  <td>
                    test
</td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "9.00",
                  },
                  "ref": null,
                  "rendered": "9.00",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "test",
                  },
                  "ref": null,
                  "rendered": "test",
                  "type": "td",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            accept=".txt"
            className="inputFile"
            id="file"
            name="file"
            onChange={[Function]}
            type="file"
/>,
          <label
            htmlFor="file"
>
            Choose a file
</label>,
          <table>
            <tbody>
                        <tr>
                                    <th>
                                                Time
                                    </th>
                                    <th>
                                                Event
                                    </th>
                        </tr>
                        <tr>
                                    <td>
                                                9.00
                                    </td>
                                    <td>
                                                test
                                    </td>
                        </tr>
            </tbody>
</table>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "accept": ".txt",
            "className": "inputFile",
            "id": "file",
            "name": "file",
            "onChange": [Function],
            "type": "file",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Choose a file",
            "htmlFor": "file",
          },
          "ref": null,
          "rendered": "Choose a file",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tbody>
              <tr>
                            <th>
                                          Time
                            </th>
                            <th>
                                          Event
                            </th>
              </tr>
              <tr>
                            <td>
                                          9.00
                            </td>
                            <td>
                                          test
                            </td>
              </tr>
</tbody>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr>
                  <th>
                                    Time
                  </th>
                  <th>
                                    Event
                  </th>
</tr>,
                Array [
                  <tr>
                    <td>
                                        9.00
                    </td>
                    <td>
                                        test
                    </td>
</tr>,
                ],
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      Time
</th>,
                    <th>
                      Event
</th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Time",
                    },
                    "ref": null,
                    "rendered": "Time",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Event",
                    },
                    "ref": null,
                    "rendered": "Event",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": "9.00",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td>
                      9.00
</td>,
                    <td>
                      test
</td>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "9.00",
                    },
                    "ref": null,
                    "rendered": "9.00",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "test",
                    },
                    "ref": null,
                    "rendered": "test",
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
            ],
            "type": "tbody",
          },
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
